admin/includes/ckeditor.php

```diff
@@ -1,40 +1,51 @@
 <?php
 /**
  * @package admin
- * @copyright Copyright 2003-2019 Zen Cart Development Team
+ * @copyright Copyright 2003-2020 Zen Cart Development Team
  * @copyright Portions Copyright 2010 Kuroi Web Design
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version $Id: Zen4All 2019 Apr 25 Modified in v1.5.6b $
+ * @version $Id:  Modified in v1.5.7 $
  */
 if (!defined('IS_ADMIN_FLAG')) {
   die('Illegal Access');
 }
-
+// prepare list of languages supported by this website, so we can tell CKEditor
 $var = zen_get_languages();
 $jsLanguageLookupArray = "var lang = new Array;\n";
 foreach ($var as $key)
 {
-  $jsLanguageLookupArray .= "  lang[" . $key['id'] . "] = '" . $key['code'] . "';\n";
+  $jsLanguageLookupArray .= "        lang[" . $key['id'] . "] = '" . $key['code'] . "';\n";
 }
 ?>
 <script>window.jQuery || document.write('<script src="https://code.jquery.com/jquery-3.4.1.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"><\/script>');</script>
 <script>window.jQuery || document.write('<script src="includes/javascript/jquery.min.js"><\/script>');</script>
-<script src="../<?php echo DIR_WS_EDITORS; ?>ckeditor/ckeditor.js"></script>
-<script>
-$(document).ready(function() {
-  <?php echo $jsLanguageLookupArray ?>
-  $('textarea').each(function() {
-    if ($(this).hasClass('editorHook') || ($(this).attr('name') != 'message' && ! $(this).hasClass('noEditor')))
-    {
-      index = $(this).attr('name').match(/\d+/);
-      if (index == null) index = <?php echo $_SESSION['languages_id'] ?>;
-      CKEDITOR.replace($(this).attr('name'),
-        {
-          coreStyles_underline : { element : 'u' },
-          width : '100%',
-          language: lang[index]
+<script src="https://cdn.ckeditor.com/4.14.0/standard-all/ckeditor.js" title="CKEditorCDN"></script>
```

Comment:
Shouldn't this be `//cdn.ckeditor.com`... instead of starting with https: as a src file if it is possible that the admin may be accessed via http: as a result of not having a SSL certificate? This would then prevent a mixed content security issue in the admin of the store...

Reply:
No, mixed content issues only happen when the base page is https: and you try to pull http: assets into it. You can pull https: assets into a http: (unsecure) page